tests/utilities/test_callables.py

```diff
@@ -297,6 +301,131 @@ def f(x: Foo):
             "type": "object",
         }
 
+    def test_function_with_pydantic_model_default_across_v1_and_v2(self):
+        # this import ensures this test imports the installed version of
+        # pydantic (not pydantic.v1) and allows us to test that we
+        # generate consistent schemas across v1 and v2
+        import pydantic
+
+        class Foo(pydantic.BaseModel):
+            bar: str
+
+        def f(foo: Foo = Foo(bar="baz")):
+            ...
+
+        schema = callables.parameter_schema(f)
+        assert schema.dict() == {
+            "title": "Parameters",
+            "type": "object",
+            "properties": {
+                "foo": {
+                    "allOf": [{"$ref": "#/definitions/Foo"}],
+                    "default": {"bar": "baz"},
+                    "position": 0,
+                    "title": "foo",
+                }
+            },
+            "definitions": {
+                "Foo": {
+                    "properties": {"bar": {"title": "Bar", "type": "string"}},
+                    "required": ["bar"],
+                    "title": "Foo",
+                    "type": "object",
+                }
+            },
+        }
+
+    def test_function_with_complex_args_across_v1_and_v2(self):
+        # this import ensures this test imports the installed version of
+        # pydantic (not pydantic.v1) and allows us to test that we
+        # generate consistent schemas across v1 and v2
+        import pydantic
+
+        class Foo(pydantic.BaseModel):
+            bar: str
+
+        def f(
+            a: int,
+            s: List[None],
+            m: Foo,
+            i: int = 0,
+            x: float = 1.0,
+            model: Foo = Foo(bar="bar"),
+            pdt: pendulum.DateTime = pendulum.datetime(2025, 1, 1),
+            pdate: pendulum.Date = pendulum.date(2025, 1, 1),
+            pduration: pendulum.Duration = pendulum.duration(seconds=5),
+        ):
+            ...
+
+        datetime_schema = {
+            "title": "pdt",
+            "default": "2025-01-01T00:00:00+00:00",
+            "position": 6,
+            "type": "string",
+            "format": "date-time",
+        }
+        duration_schema = {
+            "title": "pduration",
+            "default": 5.0,
+            "position": 8,
+            "type": "number",
+            "format": "time-delta",
+        }
+
+        if HAS_PYDANTIC_V2:
```

Comment:
These are the only remaining JsonSchema changes between pydantic v1 and v2. I think the reason is that v2 using a [standardized version of JsonSchema now](https://docs.pydantic.dev/latest/concepts/json_schema/).